File: api_CBLA/src/main/java/com/daw/persistence/entities/Jornada.java
package com.daw.persistence.entities;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "Jornada")
@Getter
@Setter
@NoArgsConstructor
public class Jornada {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer id;
	
	@Column(length = 1, nullable = false)
	private Integer semana;
	
	@Column(length = 2, nullable = false)
	private Integer mes;

	@Column(length = 1, nullable = false)
	private String fase;
	
	@Column(name = "id_temporada")
	private Integer idTemporada;

	@ManyToOne
	@JoinColumn(name = "id_temporada", referencedColumnName = "id", insertable = false, updatable = false)
	private Temporada temporada;

}

File: front_BLA/src/app/models/jornada.ts
import { Temporada } from "./temporada";

export class Jornada {
    private _id: number;
    private _semana: number;
    private _mes: number;
    private _fase: string;
    private _id_temporada: Temporada;
    


    constructor(id: number, semana: number, mes: number, fase: string, id_temporada: Temporada){
        this._id = id;
        this._semana = semana;
        this._mes = mes;
        this._fase = fase;
        this._id_temporada = id_temporada;
    }

    public get id(): number {
         return this._id;
    }

    public set id(value: number) {
        this._id = value;
    }

    public get semana(): number {
        return this._semana;
    }

    public set semana(value: number) {
        this._semana = value;
    }

    public get mes(): number {
        return this._mes;
    }

    public set mes(value: number) {
        this._mes = value;
    }

    public get fase(): string {
        return this._fase;
    }

    public set fase(value: string) {
        this._fase = value;
    }

    public get id_temporada(): Temporada {
        return this._id_temporada;
    }

    public set id_temporada(value: Temporada) {
        this._id_temporada = value;
    }
}
